src/domain/space/space/space.service.ts

```diff
@@ -1171,6 +1171,24 @@ export class SpaceService {
     return space.account;
   }
 
+  public async getSpaceForAccountOrFail(accountId: string): Promise<ISpace> {
+    const space = await this.spaceRepository.findOne({
```

Comment:
Currently, every account has a spaceID. So if we have the account then it would be more optimal to directly use its spaceID instead of iterating all the spaces.

Reply:
I have removed this one, good idea. Just passing `includeSpace` to getAccountsHostedByContributor is going to make it more efficient